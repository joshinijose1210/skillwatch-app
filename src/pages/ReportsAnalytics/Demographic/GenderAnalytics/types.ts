import { IGenderData } from '../types';

export interface Props {
    gendersData: IGenderData;
    isLoading: boolean;
}
