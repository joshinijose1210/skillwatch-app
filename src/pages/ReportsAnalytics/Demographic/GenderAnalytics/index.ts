export { GenderAnalytics as default } from './GenderAnalytics';
