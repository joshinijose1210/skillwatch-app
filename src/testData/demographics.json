{
    "gendersData": {
        "gendersCount": {
            "malesCount": 1,
            "femalesCount": 0,
            "othersCount": 13
        },
        "gendersPercentage": {
            "malesPercentage": 7.142857142857142,
            "femalesPercentage": 0.0,
            "othersPercentage": 92.85714285714286
        }
    },
    "averageTenure": {
        "years": 0,
        "months": 5
    },
    "averageAge": {
        "years": 24,
        "months": 9
    }
}
