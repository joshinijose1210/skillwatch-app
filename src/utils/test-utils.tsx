import { ToastContainer } from '@medly-components/core';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { store as rtkStore, setupStore } from '@slice';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { defaultTheme } from '@theme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import reduxMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { userMockState } from '../mocks/userMockState';
import { initialState } from '../slice';

interface CustomOptions extends RenderOptions {
    initialState?: any;
    initialEntries?: any;
}

interface WithThemeProviderProps {
    children?: React.ReactNode;
}

const mockStore = reduxMockStore(),
    store = mockStore({
        ...initialState,
        user: { ...initialState.user, groups: ['admin'] }
    });

const WithThemeProvider: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>
        <>
            <ToastContainer position="top-end" />
            {children}
        </>
    </ThemeProvider>
);

const WithStore: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <WithThemeProvider>{children}</WithThemeProvider>
    </Provider>
);

const WithRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <MemoryRouter>
        <WithThemeProvider>{children}</WithThemeProvider>
    </MemoryRouter>
);

const WithStoreAndRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <MemoryRouter>
            <WithThemeProvider>{children}</WithThemeProvider>
        </MemoryRouter>
    </Provider>
);

export const RTKRender = (ui: React.ReactElement, { initialState = userMockState, ...renderOptions }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);

    const RtkWrapper: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={renderOptions.initialEntries}>
                <WithThemeProvider>{children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
    return { rtkStore, ...render(ui, { wrapper: RtkWrapper, ...renderOptions }) };
};
const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithThemeProvider, ...options });

export const renderWithStore = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStore, ...options });

export const renderWithRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithRouter, ...options });

export const renderWithStoreAndRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStoreAndRouter, ...options });

export const wrapper = (props: { children: React.ReactNode }, { initialState = userMockState }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);
    return (
        <Provider store={store}>
            <MemoryRouter>
                <WithThemeProvider>{props.children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render, store };
