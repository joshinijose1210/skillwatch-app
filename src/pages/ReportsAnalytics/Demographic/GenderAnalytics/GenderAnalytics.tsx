import { Box, Text } from '@medly-components/core';
import { DotsLoader } from '@medly-components/loaders';
import { FC, memo } from 'react';
import { ReactComponent as FemaleIcon } from '../../../../constants/images/icons/femaleIcon.svg';
import { ReactComponent as MaleIcon } from '../../../../constants/images/icons/maleIcon.svg';
import { ReactComponent as OtherGenderIcon } from '../../../../constants/images/icons/otherGenderIcon.svg';
import { GenderAnalyticsSection, GenderContent, GenderContentWrapper, DemographicLabel, StyledDivider } from '../Demographic.styled';
import { Props } from './types';

export const GenderAnalytics: FC<Props> = memo(({ gendersData, isLoading }) => {
    return (
        <GenderAnalyticsSection>
            <GenderContentWrapper>
                <FemaleIcon className="female" />
                <GenderContent>
                    {isLoading ? (
                        <DotsLoader />
                    ) : (
                        <Box display="flex" bg="inherit">
                            <Text textColor="#333" textWeight="Strong" className="value">
                                {gendersData?.gendersCount?.femalesCount ?? 0}
                            </Text>
                            <StyledDivider textWeight="Light" className="value">
                                |
                            </StyledDivider>
                            <Text textColor="#333" textWeight="Strong" className="value">
                                {gendersData?.gendersPercentage?.femalesPercentage?.toFixed(1) ?? 0}%
                            </Text>
                        </Box>
                    )}
                    <DemographicLabel>Female</DemographicLabel>
                </GenderContent>
            </GenderContentWrapper>
            <GenderContentWrapper>
                <MaleIcon className="male" />
                <GenderContent>
                    {isLoading ? (
                        <DotsLoader />
                    ) : (
                        <Box display="flex" bg="inherit">
                            <Text textColor="#333" textWeight="Strong" className="value">
                                {gendersData?.gendersCount?.malesCount ?? 0}
                            </Text>
                            <StyledDivider textWeight="Light" className="value">
                                |
                            </StyledDivider>
                            <Text textColor="#333" textWeight="Strong" className="value">
                                {gendersData?.gendersPercentage?.malesPercentage?.toFixed(1) ?? 0}%
                            </Text>
                        </Box>
                    )}
                    <DemographicLabel>Male</DemographicLabel>
                </GenderContent>
            </GenderContentWrapper>
            <GenderContentWrapper>
                <OtherGenderIcon className="other" />
                <GenderContent>
                    {isLoading ? (
                        <DotsLoader />
                    ) : (
                        <Box display="flex" bg="inherit">
                            <Text textColor="#333" textWeight="Strong" className="value">
                                {gendersData?.gendersCount?.othersCount ?? 0}
                            </Text>
                            <StyledDivider textWeight="Light" className="value">
                                |
                            </StyledDivider>
                            <Text textColor="#333" textWeight="Strong" className="value">
                                {gendersData?.gendersPercentage?.othersPercentage?.toFixed(1) ?? 0}%
                            </Text>
                        </Box>
                    )}
                    <DemographicLabel>Other</DemographicLabel>
                </GenderContent>
            </GenderContentWrapper>
        </GenderAnalyticsSection>
    );
});

GenderAnalytics.displayName = 'GenderAnalytics';
