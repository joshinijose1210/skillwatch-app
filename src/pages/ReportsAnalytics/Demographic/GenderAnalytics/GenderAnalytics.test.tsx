import { RTKRender } from '@utils/test-utils';
import { demographicData } from '@mocks/demographics';
import { GenderAnalytics } from './GenderAnalytics';

describe('GenderAnalytics', () => {
    it('should render properly', () => {
        const { container } = RTKRender(<GenderAnalytics isLoading={false} gendersData={demographicData.gendersData} />);
        expect(container).toMatchSnapshot();
    });
});
