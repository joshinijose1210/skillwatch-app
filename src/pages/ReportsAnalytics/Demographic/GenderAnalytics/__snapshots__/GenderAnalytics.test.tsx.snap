// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenderAnalytics should render properly 1`] = `
<div>
  <div
    class="sc-llmrdn-1 dKJLYy"
  >
    <div
      class="sc-llmrdn-3 kjWyQI"
    >
      <svg
        class="female"
        fill="#333"
        height="61"
        viewBox="0 0 27 61"
        width="27"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M26.765 42.89h-9.962v17.893H9.994V42.912H0c.224-.675.417-1.286.63-1.898 2.352-6.925 4.704-13.85 7.077-20.776.887-2.584 2.715-4.02 5.419-4.202a5.785 5.785 0 0 1 5.847 3.902c1.464 4.224 2.897 8.47 4.329 12.704 1.112 3.27 2.213 6.529 3.324 9.798.043.14.086.28.14.45zM13.383.447c3.292 0 5.985 2.68 5.975 5.95-.011 3.302-2.683 5.982-5.954 5.982-3.303 0-5.997-2.712-5.975-6.003A5.955 5.955 0 0 1 13.383.447z"
          fill="#000"
        />
      </svg>
      <div
        class="sc-llmrdn-4 gQBmvU"
      >
        <div
          bg="inherit"
          borderradius="2"
          class="sc-12ecrgm-0 eQSGuc"
          color="black"
          display="flex"
          font-variant="body2"
          px="4"
          py="4"
        >
          <strong
            class="sc-1ntgl3l-0 hSccqi value"
            textalign="initial"
            textcolor="#333"
            textweight="Strong"
          >
            0
          </strong>
          <span
            class="sc-1ntgl3l-0 ezeYxn sc-llmrdn-6 eZufyN value"
            textalign="initial"
            textcolor="inherit"
            textweight="Light"
          >
            |
          </span>
          <strong
            class="sc-1ntgl3l-0 hSccqi value"
            textalign="initial"
            textcolor="#333"
            textweight="Strong"
          >
            0.0
            %
          </strong>
        </div>
        <span
          class="sc-1ntgl3l-0 icUogw sc-llmrdn-8 YrKSM"
          textalign="initial"
          textcolor="#666"
        >
          Female
        </span>
      </div>
    </div>
    <div
      class="sc-llmrdn-3 kjWyQI"
    >
      <svg
        class="male"
        fill="#333"
        height="60"
        viewBox="0 0 18 60"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_1346_9876)"
        >
          <path
            d="M17.99 37.516h-5.294V60H5.315V37.516H.02c0-.162-.021-.302-.021-.431V21.147c0-3.02.833-5.403 3.65-6.018.39-.086.79-.118 1.192-.118C7.617 15 10.392 15 13.168 15c2.879 0 4.133 1.952 4.709 4.907.082.398.113.808.113 1.207v16.402z"
            fill="#000"
          />
        </g>
        <path
          d="M8.383.447c3.292 0 5.985 2.68 5.975 5.95-.011 3.302-2.683 5.982-5.954 5.982-3.303 0-5.997-2.712-5.975-6.003A5.955 5.955 0 0 1 8.383.447z"
          fill="#000"
        />
        <defs>
          <clippath
            id="clip0_1346_9876"
          >
            <path
              d="M0 0h18v45H0z"
              fill="#fff"
              transform="translate(0 15)"
            />
          </clippath>
        </defs>
      </svg>
      <div
        class="sc-llmrdn-4 gQBmvU"
      >
        <div
          bg="inherit"
          borderradius="2"
          class="sc-12ecrgm-0 eQSGuc"
          color="black"
          display="flex"
          font-variant="body2"
          px="4"
          py="4"
        >
          <strong
            class="sc-1ntgl3l-0 hSccqi value"
            textalign="initial"
            textcolor="#333"
            textweight="Strong"
          >
            1
          </strong>
          <span
            class="sc-1ntgl3l-0 ezeYxn sc-llmrdn-6 eZufyN value"
            textalign="initial"
            textcolor="inherit"
            textweight="Light"
          >
            |
          </span>
          <strong
            class="sc-1ntgl3l-0 hSccqi value"
            textalign="initial"
            textcolor="#333"
            textweight="Strong"
          >
            7.1
            %
          </strong>
        </div>
        <span
          class="sc-1ntgl3l-0 icUogw sc-llmrdn-8 YrKSM"
          textalign="initial"
          textcolor="#666"
        >
          Male
        </span>
      </div>
    </div>
    <div
      class="sc-llmrdn-3 kjWyQI"
    >
      <svg
        class="other"
        fill="none"
        height="60"
        viewBox="0 0 25 60"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.692 41.085h10.677L16.8 19.402c-.799-1.86-3-3.87-7.8-3.87s-2.428 2.01-.642 3.016c-.779 7.542-.001 16.09-.27 22.123v1.005h-.01v18.101h5.615V41.085z"
          fill="#000"
        />
        <path
          d="M4.815 41.18v-.5H.5V21.7c0-1.483.205-2.746.71-3.698.491-.925 1.28-1.586 2.547-1.864.347-.077.71-.108 1.085-.108h3.647L7.51 59.277H4.815V41.18z"
          fill="#000"
          stroke="#000"
        />
        <path
          d="M8.383.447c3.292 0 5.985 2.68 5.975 5.95-.011 3.302-2.683 5.982-5.954 5.982-3.303 0-5.997-2.712-5.975-6.003A5.955 5.955 0 0 1 8.383.447z"
          fill="#000"
        />
        <path
          d="M8 15h2v45H8V15z"
          fill="#F7F9FD"
        />
      </svg>
      <div
        class="sc-llmrdn-4 gQBmvU"
      >
        <div
          bg="inherit"
          borderradius="2"
          class="sc-12ecrgm-0 eQSGuc"
          color="black"
          display="flex"
          font-variant="body2"
          px="4"
          py="4"
        >
          <strong
            class="sc-1ntgl3l-0 hSccqi value"
            textalign="initial"
            textcolor="#333"
            textweight="Strong"
          >
            13
          </strong>
          <span
            class="sc-1ntgl3l-0 ezeYxn sc-llmrdn-6 eZufyN value"
            textalign="initial"
            textcolor="inherit"
            textweight="Light"
          >
            |
          </span>
          <strong
            class="sc-1ntgl3l-0 hSccqi value"
            textalign="initial"
            textcolor="#333"
            textweight="Strong"
          >
            92.9
            %
          </strong>
        </div>
        <span
          class="sc-1ntgl3l-0 icUogw sc-llmrdn-8 YrKSM"
          textalign="initial"
          textcolor="#666"
        >
          Other
        </span>
      </div>
    </div>
  </div>
</div>
`;
